lib/widgets/forest_park_map.dart

```diff
@@ -73,10 +80,16 @@ class _ForestParkMapState extends ConsumerState<ForestParkMap> with WidgetsBindi
     // final parkTrails = ref.watch(parkTrailsProvider);
     final locationStatus = ref.watch(locationPermissionStatusProvider);
 
-    final followOnLocation = ref.watch(followOnLocationProvider);
-    ref.listen(followOnLocationProvider, (prev, next) {
-      if (next != prev && next != FollowOnLocationUpdate.never) {
-        _followCurrentLocationStreamController.add(null);
+    final followOnLocation = ref.watch(followOnLocationTargetProvider);
+    ref.listen(followOnLocationTargetProvider, (prev, next) {
+      switch (next) {
+        case FollowOnLocationTargetState.none:
+          _followCurrentLocationStreamController.add(null);
+        case FollowOnLocationTargetState.currentLocation:
+          ref.read(followOnLocationTargetProvider.notifier).state = FollowOnLocationTargetState.forestPark;
```

Comment:
Setting the followOnLocationTargetProvider to forestPark in the listener means that as soon as someone tries to center on currentLocation, it'll jump to forestPark.
After testing, adding null to the _followCurrentLocationStreamController causes some stutters when moving the map when following the current location (make sure you're testing for functionality while you're working!)
That means the only needed case is for .forestPark, which can be replaced with an if statement.